lib/node/http.js

```diff
@@ -165,6 +166,37 @@ class HTTP extends Server {
       });
     });
 
+    // Balance by address
+    this.get('/balance/:address', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const address = valid.str('address');
+
+      enforce(address, 'Address is required.');
+      enforce(!this.chain.options.spv, 'Cannot get balance in SPV mode.');
+
+      const addr = Address.fromString(address, this.network);
+      const coins = await this.node.getCoinsByAddress(addr);
+      const result = {
+        height: this.chain.height,
+        balance: 0,
+        coins: [],
+        lockedBalance: 0,
+        lockedCoins: []
+      };
+
+      for (const coin of coins) {
+        if (coin.covenant.type === rules.types.NONE) {
```

Comment:
You could save needing to import `lib/covenant/rules` by using `covenant.isNone` 

https://github.com/handshake-org/hsd/blob/cb182f47fe9a6d14aa5b6addb59b214dbff705b2/lib/primitives/covenant.js#L234